stdlib/src/collections/list.mojo

```diff
@@ -497,12 +497,72 @@ struct List[T: CollectionElement, hint_trivial_type: Bool = False](
 
         Args:
             value: The value to append.
+
+        Notes:
+            If there is no capacity left, resizes to twice the current capacity.
+            Except for 0 capacity where it sets 1.
         """
         if self.size >= self.capacity:
-            self._realloc(max(1, self.capacity * 2))
+            self._realloc(self.capacity * 2 + int(self.capacity == 0))
```

Comment:
**Comment** Nice, this is a clever little optimization! 🧠

Reply:
Hi @soraros. Looking at the assembly it might seem so.

```assembly
.Lf$local:
	leaq	(%rdi,%rdi), %rax ; %rax = %rdi * %rdi
	cmpq	$1, %rax ; compare and set the flags register
	adcq	$0, %rax ; %rax = %rax + $0 + carry_flag
	retq
```
```assembly
.Lg$local:
	xorl	%eax, %eax ; set %eax to zero
	testq	%rdi, %rdi ; set the flags to 1 if %rdi is zero, no-op
	sete	%al ; %al = int(flags == 1)
	leaq	(%rax,%rdi,2), %rax ; %rax = %rax + %rdi * 2
	retq
```
I'm not sure which one is faster just by looking at this. The lower one seems to involve more registers and ops, but I don't know how this would look pipelined at runtime. I think the `testq` and `sete` are practically no-ops in this case, but I'm not sure since I don't program in assembly much

It might be fixed by doing 
```mojo
var double = UInt(self.capacity) * 2
self._realloc(double + int(double == 0))
```